src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs

```diff
@@ -133,12 +165,536 @@ public async Task CreateInvitation_AsViewer_ShouldReturnForbidden_WhenPermission
+                UserType.Planner,
+                TestFactory.UnknownPlant,
+                9999,
+                new EditInvitationDto(),
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task EditInvitation_AsHacker_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.EditInvitationAsync(
+                UserType.Hacker,
+                TestFactory.PlantWithAccess,
+                9999,
+                new EditInvitationDto(),
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task EditInvitation_AsPlanner_ShouldReturnBadRequest_WhenUnknownInvitationId()
+        {
+            var editInvitationDto = await CreateValidEditInvitationDto();
+
+            await InvitationsControllerTestsHelper.EditInvitationAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                38934,
+                editInvitationDto,
+                HttpStatusCode.BadRequest);
+        }
+
+        private async Task<EditInvitationDto> CreateValidEditInvitationDto()
+        {
+            var invitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                InitialInvitationId);
+
+            invitation.Status = IpoStatus.Planned;
 
+            var editInvitationDto = new EditInvitationDto
+            {
+                Title = invitation.Title,
+                Description = invitation.Description,
+                StartTime = invitation.StartTimeUtc,
+                EndTime = invitation.EndTimeUtc,
+                Location = invitation.Location,
+                ProjectName = invitation.ProjectName,
+                RowVersion = invitation.RowVersion,
+                UpdatedParticipants = ConvertToParticipantDtoEdit(invitation.Participants),
+                UpdatedCommPkgScope = null,
+                UpdatedMcPkgScope = _mcPkgScope
+            };
+
+            return editInvitationDto;
+        }
+
+        private IEnumerable<ParticipantDtoEdit> ConvertToParticipantDtoEdit(IEnumerable<ParticipantDtoGet> participants)
+        {
+            var editVersionParticipantDtos = new List<ParticipantDtoEdit>();
+            participants.ToList().ForEach(p => editVersionParticipantDtos.Add(
+                new ParticipantDtoEdit
+                {
+                    ExternalEmail = p.ExternalEmail,
+                    FunctionalRole = p.FunctionalRole,
+                    Organization = p.Organization,
+                    Person = p.Person?.Person,
+                    SortKey = p.SortKey
+                }));
+
+            return editVersionParticipantDtos;
+        }
         #endregion
 
-        #region Edit (Sign)
-        //todo add tests 
+        #region Sign 
```

Comment:
I think more negative tests regarding permissions should be added:
* Sign should fail if planner (after removing SIGN from planners permissions list)